internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -343,7 +380,7 @@ func (q amsQuotaService) ReserveQuota(kafka *dbapi.KafkaRequest, instanceType ty
 		ExternalClusterID(kafkaId).
 		Disconnected(false).
 		BYOC(false).
-		AvailabilityZone("single").
+		AvailabilityZone(q.getAMSClusterAuthorizationRequestAvailabilityZone(kafka.MultiAZ)).
```

Comment:
I noticed that aside from in the AMS `ReservedResourceBuilder` the AvailabilityZone is also set for the ClusterAuthorization request. I updated the logic to also set it depending on the kafka request multiaz value

Reply:
cc @machi1990

Reply:
After further talks with AMS team we've been told that the behavior is that the AvailabilityZone attribute has priority at ClusterAuthorizationRequest level over at ReservedResource level, overwritting it's content.
Thus, it's not needed to set at both levels. Same applies to the BYOC parameter.
I've removed the setting of the BYOC and AvailabilityZoneType attributes from the ReservedResource and set them at ClusterAuthorizationRequest level only so we don't have duplicate setting of those attributes.

Reply:
> I've removed the setting of the BYOC and AvailabilityZoneType attributes from the ReservedResource and set them at ClusterAuthorizationRequest level only so we don't have duplicate setting of those attributes.

Isn't it clear codewise to set it appropriately for the `ReservedResource` as well? 

Someone can easily get confused to see it set to `multi` for `ClusterAuthorizationRequest` and `single` for the `ReservedResource` for the same Multi_Az Kafka.

Reply:
> Isn't it clear codewise to set it appropriately for the ReservedResource as well?



> > I've removed the setting of the BYOC and AvailabilityZoneType attributes from the ReservedResource and set them at ClusterAuthorizationRequest level only so we don't have duplicate setting of those attributes.
> 
> Isn't it clear codewise to set it appropriately for the `ReservedResource` as well?
> 
> Someone can easily get confused to see it set to `multi` for `ClusterAuthorizationRequest` and `single` for the `ReservedResource` for the same Multi_Az Kafka.

ClusterAuthorizationRequest would overwrite what's in the ReservedResource. We've been told that we should set it at ClusterAuthorizationRequest level

Reply:
> Someone can easily get confused to see it set to `multi` for `ClusterAuthorizationRequest` and `single` for the `ReservedResource` for the same Multi_Az Kafka. 

That does not happen

Reply:
Thanks for the clarification